apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/SamhsaMatcherR4FromClaimTransformerV2Test.java

```diff
@@ -0,0 +1,845 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import static org.junit.Assert.assertEquals;
+
+import com.codahale.metrics.MetricRegistry;
+import gov.cms.bfd.model.codebook.data.CcwCodebookVariable;
+import gov.cms.bfd.model.rif.CarrierClaim;
+import gov.cms.bfd.model.rif.DMEClaim;
+import gov.cms.bfd.model.rif.HHAClaim;
+import gov.cms.bfd.model.rif.HospiceClaim;
+import gov.cms.bfd.model.rif.InpatientClaim;
+import gov.cms.bfd.model.rif.OutpatientClaim;
+import gov.cms.bfd.model.rif.PartDEvent;
+import gov.cms.bfd.model.rif.RifRecordBase;
+import gov.cms.bfd.model.rif.SNFClaim;
+import gov.cms.bfd.model.rif.samples.StaticRifResourceGroup;
+import gov.cms.bfd.server.war.ServerTestUtils;
+import gov.cms.bfd.server.war.commons.CCWUtils;
+import gov.cms.bfd.server.war.commons.IcdCode;
+import gov.cms.bfd.server.war.commons.TransformerConstants;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.ExplanationOfBenefit;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/** Verifies that transformations that contain SAMHSA codes are filtered as expected. */
+@RunWith(Parameterized.class)
+public class SamhsaMatcherR4FromClaimTransformerV2Test {
+
+  private R4SamhsaMatcher samhsaMatcherV2;
+  private static final String CODING_SYSTEM_HCPCS_CD =
+      TransformerConstants.BASE_URL_CCW_VARIABLES
+          + "/"
+          + CcwCodebookVariable.HCPCS_CD.getVariable().getId().toLowerCase();
+  private static final String DRG_SYSTEM =
+      CCWUtils.calculateVariableReferenceUrl(CcwCodebookVariable.CLM_DRG_CD);
+  private static final String BLACKLISTED_HCPCS_CODE = "M1034";
+  private static final String NON_SAMHSA_HCPCS_CODE = "11111";
+  private static final String BLACKLISTED_IC9_DIAGNOSIS_CODE = "291.0";
+  private static final String BLACKLISTED_IC10_DIAGNOSIS_CODE = "F10.10";
+  private static final String BLACKLISTED_IC9_PROCEDURE_CODE = "94.45";
+  private static final String BLACKLISTED_IC10_PROCEDURE_CODE = "HZ2ZZZZ";
+  private static final String NON_BLACKLISTED_IC_CODE = "111111";
+  private static final String BLACKLISTED_DRG_DIAGNOSIS_CODE = "522";
+  private static final String NON_BLACKLISTED_DRG_DIAGNOSIS_CODE = "1111111";
+
+  private final RifRecordBase claim;
+  private final ExplanationOfBenefit loadedExplanationOfBenefit;
+
+  /** Sets up the test. */
+  @Before
+  public void setup() {
+    samhsaMatcherV2 = new R4SamhsaMatcher();
+  }
+
+  /**
+   * Data collection.
+   *
+   * @return the collection
+   */
+  @Parameterized.Parameters(name = "{0}")
```

Comment:
.class doesn't seem to work, the parameterized namestring seems pretty limited. It can only use one of the parameters' toString, so to make it prettier i'd have to format the classname and pass a separate list of params just for test display purposes. At the moment it uses the ugly full path+classname which was a good improvement over the default of just using the parameter index, and was the best thing I could get to work without doing something hacky(er) just for display purposes:

![image](https://user-images.githubusercontent.com/1200656/140828100-d513dcff-4eea-4074-87ca-27c3417a556b.png)

Reply:
I had a small issue with this where from the test result view, if I tried to right click and rerun a test, it was passing this reference from the previous run and the test was failing. Given that plus the conversation below about the claim not being needed, I'm leaning slightly toward the unused parameter solution so that the name of the test would be things like 'INPATIENT', 'HHA', etc.

Reply:
I don't think it's great to have the test case names change every time, as it could confuse things that track test case successes vs. failures over time, e.g. Jenkins.

This is still a minor point that shouldn't block approval, but I'd add a new name param as `{0}` in the params array to address this.

Reply:
Wrote up an adjustment to pass the output from the util, and confirmed with the fix that the tests can be rerun individually:
![image](https://user-images.githubusercontent.com/1200656/140965709-1448f30f-db71-49a5-a6b3-6ac923ec8b98.png)